### PR TITLE
Feature/smar78 and 84 create and update patient

### PR DESCRIPTION
**Pull request by SwiftXLight**
**Description changes:** 
- add new entity and dto for patient
- add create and patch patient endpoint
- add get patients endpoint
- add guards for endpoints

[link to Jira ticket 78](https://smartdc23.atlassian.net/browse/SMAR-78?atlOrigin=eyJpIjoiMDUwNGJkYzQ5NzM4NGMzYWFmN2JkNGIyMTJjMDM1ODEiLCJwIjoiaiJ9)
[link to Jira ticket 84](https://smartdc23.atlassian.net/browse/SMAR-84?atlOrigin=eyJpIjoiN2U0NmNiZTA0OTBmNGE4YmE4ODk0YTg1NGUzMGNkNmYiLCJwIjoiaiJ9)

**PR checklist:**
1.1 Common
1.1.1 types for input and output params
1.1.2 no any, should be also added to eslint rules
1.1.3 try/catch for any async function
1.1.4 no magic numbers
1.1.5 compare only with constants not with strings
(instead of user === ‘admin’, better user === roles.admin)
1.1.6 no ternary operator inside ternary operator (bad example: question ? first:
second ? third: fourth)
1.1.7 avoid similar types with duplicating by generics
1.1.8 no console.log in pr
1.1.9 .env file should be in .gitignore
1.1.10 delete commented code if it's not part of documentation
1.1.11 no links in the code, env links should be in env file (for example: server url),
constant links (for example default avatar URL) should be in constant file
1.1.12 constants and function names should be lowercase.
1.1.13 no dates format in the code (like "dd/MM/yyyy”), move it in global constant
variable
1.1.14 import rules. imports should come in a specific order: node modules, absolute
path, relative path
1.1.15 strings should be in the constant variable. Example: instead of ‘15 3 * * *’,
const EACH_DAY_15h_15min = ‘15 3 * * *’

1.2 Backend
1.2.1 swagger for each endpoint
1.2.2 less requests to db, all data should be taken with one query
1.2.3 `public` in method use only is use function externally
1.2.4 use ConfigService instead of `process.env`

Create new 'patient' entity:
image

Update patient info:
image

Try to update patient without token:
image

Create new patient:
image

Try to create patient without token:
image

### CODE REVIEW

File: src/modules/patient/patient.service.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    try {
+      return await this.patientRepository
+        .createQueryBuilder('patient')
+        .getMany();
+    } catch (err) {
+      throw new HttpException(`${err}`, HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+  }
+
+  async createPatient(createPatientDto: CreatePatientDto): Promise<Patient> {
```
Comment (lyudad) [inline src/modules/patient/patient.service.ts]: async createPatient(patient: CreatePatientDto)

File: src/modules/patient/patient.service.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+      return newPatient.generatedMaps[0] as Patient;
+    } catch (err) {
+      throw new HttpException(`${err}`, HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+  }
+
+  async updatePatient(
+    id: number,
+    patientDto: Partial<CreatePatientDto>,
```
Comment (lyudad) [inline src/modules/patient/patient.service.ts]: patient: Partial<CreatePatientDto>,

File: src/modules/patient/patient.service.ts (hunk 1 above)
Comment (SwiftXLight) [reply src/modules/patient/patient.service.ts]: fixed

File: src/modules/patient/patient.service.ts (hunk 2 above)
Comment (SwiftXLight) [reply src/modules/patient/patient.service.ts]: fixed

**Pull request merged**